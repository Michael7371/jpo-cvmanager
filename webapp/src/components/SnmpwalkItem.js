import React from 'react'

import './css/SnmpItem.css'

const SnmpwalkItem = (props) => {
  return (
    <div id="snmpitemdiv">
      <h3 id="snmpitemheader">{props.index}</h3>
      <p id="snmpitemtext">
        <strong>Message Type:</strong> {props.content['Message Type']}
      </p>
      <p id="snmpitemtext">
        <strong>Destination IP:</strong> {props.content['IP']}
      </p>
      <p id="snmpitemtext">
        <strong>Port:</strong> {props.content['Port']}
      </p>
      <p id="snmpitemtext">
        <strong>Protocol:</strong> {props.content['Protocol']}
      </p>
      <p id="snmpitemtext">
        <strong>Start:</strong> {props.content['Start DateTime']}
      </p>
      <p id="snmpitemtext">
        <strong>End:</strong> {props.content['End DateTime']}
      </p>
      <p id="snmpitemtext">
        <strong>Active:</strong> {props.content['Config Active']}
      </p>
    </div>
  )
}

export default SnmpwalkItem
