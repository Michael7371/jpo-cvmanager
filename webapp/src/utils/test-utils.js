/**
 * This function searches for the every somewhat random/chaotic class name and property that cause snapshot tests to be inconsistent.
 * This current list includes MUI classes (css-*) and aria-invalid attributes.
 *
 * @ex :
 * ```
 * const { container } = render(<Component />);
 *
 * replaceChaoticIds(container);
 * ```
 *
 * @param container The HTMLElement node to search for SSR ids
 */
function replaceChaoticIds(container) {
  const props = [
    {
      selector: 'class',
      updateFunc: (val) => val.replace(/css-[0-9a-z]*?-/g, 'css-mocked-'),
    },
  ]

  container.querySelectorAll('*').forEach((item) => {
    props.forEach((prop) => {
      if (item.getAttribute(prop.selector)) {
        item.setAttribute(prop.selector, prop.updateFunc(item.getAttribute(prop.selector)))
      }
    })
  })
  container.querySelectorAll('input[aria-invalid]').forEach((item) => {
    item.removeAttribute('aria-invalid')
  })
  return container
}

export { replaceChaoticIds }
