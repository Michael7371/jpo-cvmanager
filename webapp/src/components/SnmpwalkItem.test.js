import React from 'react'
import { render } from '@testing-library/react'
import SnmpwalkItem from './SnmpwalkItem'
import { replaceChaoticIds } from '../utils/test-utils'

it('should take a snapshot', () => {
  const { container } = render(<SnmpwalkItem content={{}} />)

  expect(replaceChaoticIds(container)).toMatchSnapshot()
})
