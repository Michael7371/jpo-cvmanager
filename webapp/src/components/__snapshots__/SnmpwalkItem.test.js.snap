// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<div>
  <div
    id="snmpitemdiv"
  >
    <h3
      id="snmpitemheader"
    />
    <p
      id="snmpitemtext"
    >
      <strong>
        Message Type:
      </strong>
       
    </p>
    <p
      id="snmpitemtext"
    >
      <strong>
        Destination IP:
      </strong>
       
    </p>
    <p
      id="snmpitemtext"
    >
      <strong>
        Port:
      </strong>
       
    </p>
    <p
      id="snmpitemtext"
    >
      <strong>
        Protocol:
      </strong>
       
    </p>
    <p
      id="snmpitemtext"
    >
      <strong>
        Start:
      </strong>
       
    </p>
    <p
      id="snmpitemtext"
    >
      <strong>
        End:
      </strong>
       
    </p>
    <p
      id="snmpitemtext"
    >
      <strong>
        Active:
      </strong>
       
    </p>
  </div>
</div>
`;
